components/brave_vpn/resources/panel/container.tsx

```diff
@@ -42,7 +52,14 @@ function Main () {
     apiProxy.getInstance().addVPNObserver({
       onConnectionCreated: () => {/**/},
       onConnectionRemoved: () => {/**/},
-      onConnectionStateChanged: (state: ConnectionState) => setStatus(state)
+      onConnectionStateChanged: (state: ConnectionState) => {
+        if (state === ConnectionState.CONNECT_FAILED) {
+          setHasError(true)
+          setOn(false)
+        }
+
+        setStatus(state)
```

Comment:
I think we had talked here about not setting these state variables, since they do not contain any extra information for the renderer other than `status === ConnectionState.CONNECT_FAILED`. But if you think there's a reason to go ahead with it like this, I'm fine with it.